File: dotnet/cafdemalihapa/Directories/Hosting/Feature/BackEnd/BackEndPrimary/BackEndPrimaryDirectory.cs
using cafdemalihapa.Names;

namespace cafdemalihapa.Directories.Hosting.Feature.BackEnd.BackEndPrimary
{
    public class BackEndPrimaryDirectory(
            BackEndDirectory backEndDirectory,
            PrimaryApplication primaryApplication
        )
    {
        public string GetName()
        {
            var backEndDirectoryName = backEndDirectory.GetName();
            var primaryApplicationName = primaryApplication.GetName();
            var name = $"{backEndDirectoryName}.{primaryApplicationName}";
            return name;
        }

        public string GetPath()
        {
            var backEndDirectoryPath = backEndDirectory.GetPath();
            var name = GetName();

            var backEndPrimaryDirectoryPath = Path.Combine(backEndDirectoryPath, name);
            return backEndPrimaryDirectoryPath;
        }

        public void Create()
        {
            var path = GetPath();
            if(!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/BackEnd/BackEndPrimary/BackEndPrimaryDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.BackEnd.BackEndPrimary

open System.IO
open cafdemwimu.console.Names
open cafdemwimu.console.Directories.Hosting.Feature.BackEnd

type BackEndPrimaryDirectory(primaryApplication: PrimaryApplication) =
    member _.GetName() =
        let backEndDirectoryName = BackEndDirectory.GetName()
        let primaryApplicationName = primaryApplication.GetName()
        let name = $"{backEndDirectoryName}.{primaryApplicationName}"
        name

    member this.GetPath() =
        let backEndDirectoryPath = BackEndDirectory.GetPath()
        let name = this.GetName()
        let backEndPrimaryDirectoryPath = Path.Combine(backEndDirectoryPath, name)
        backEndPrimaryDirectoryPath

    member this.Create() =
        let path = this.GetPath()
        if not (Directory.Exists(path)) then
            Directory.CreateDirectory(path) |> ignore
